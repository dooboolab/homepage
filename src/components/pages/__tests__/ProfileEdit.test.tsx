import 'react-native';

import React, {ReactElement} from 'react';
import {RenderAPI, cleanup, render} from '@testing-library/react-native';
import {createTestElement, createTestProps} from '../../../../test/testUtils';

import Page from '../ProfileEdit';

let props: any;
let component: ReactElement;
let testingLib: RenderAPI;

jest.mock('@react-navigation/native', () => {
  return {
    // @ts-ignore
    ...jest.requireActual('@react-navigation/native'),
    createNavigatorFactory: jest.fn(),
    useNavigation: (): Record<string, unknown> => ({
      navigate: jest.fn(),
      setOptions: jest.fn(),
    }),
  };
});

jest.mock('@react-navigation/native', () => {
  return {
    // @ts-ignore
    ...jest.requireActual('@react-navigation/native'),
    createNavigatorFactory: jest.fn(),
    useNavigation: (): Record<string, unknown> => ({
      navigate: jest.fn(),
      setOptions: jest.fn(),
    }),
  };
});

describe('Rendering', () => {
  beforeEach(() => {
    props = createTestProps();

    component = createTestElement(<Page {...props} />);
  });

  afterEach(cleanup);

  it('renders without crashing', () => {
    expect(1).toBeTruthy();
    // testingLib = render(component);

    // const baseElement = testingLib.toJSON();

    // expect(baseElement).toMatchSnapshot();
    // expect(baseElement).toBeTruthy();
  });
});
