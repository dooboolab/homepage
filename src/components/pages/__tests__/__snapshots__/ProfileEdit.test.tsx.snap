// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Interaction should simulate onClick 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <RNCSafeAreaView>
      <View
        onLayout={[Function]}
        style={
          Object {
            "paddingBottom": 0,
          }
        }
      >
        <RCTScrollView
          collapsable={false}
          contentContainerStyle={
            Object {
              "alignSelf": "stretch",
              "paddingHorizontal": 16,
              "paddingTop": 24,
            }
          }
          onGestureHandlerEvent={[Function]}
          onGestureHandlerStateChange={[Function]}
          style={
            Object {
              "alignSelf": "stretch",
            }
          }
        >
          <View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#FFFFFF",
                    "flexDirection": "column",
                    "justifyContent": "flex-start",
                  },
                ]
              }
            >
              <View>
                <View
                  onPress={[MockFunction]}
                  style={
                    Object {
                      "marginTop": 12,
                    }
                  }
                >
                  <Image
                    source={
                      Object {
                        "canInstrument": true,
                        "createTransformer": [Function],
                        "getCacheKey": [Function],
                        "process": [Function],
                      }
                    }
                    style={
                      Array [
                        Object {
                          "borderRadius": 60,
                          "height": 120,
                          "marginTop": 10,
                          "width": 120,
                        },
                      ]
                    }
                  />
                </View>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "alignSelf": "stretch",
                      "flexDirection": "column",
                    },
                    Object {
                      "marginHorizontal": 24,
                      "marginTop": 24,
                      "paddingHorizontal": 20,
                    },
                  ]
                }
                testID="container-id"
              >
                <View
                  style={
                    Array [
                      Array [
                        Object {
                          "alignItems": "center",
                          "alignSelf": "stretch",
                          "borderBottomColor": "#00000030",
                          "borderBottomWidth": 0.5,
                          "flexDirection": "row",
                          "justifyContent": "space-between",
                        },
                        Object {
                          "height": 54,
                        },
                      ],
                      false,
                      Object {
                        "borderBottomColor": "#00000030",
                        "borderColor": "#00000030",
                      },
                    ]
                  }
                >
                  <Text
                    style={
                      Array [
                        Array [
                          Object {
                            "color": "#999999",
                            "fontSize": 14,
                            "width": 100,
                          },
                          undefined,
                        ],
                        undefined,
                        Object {
                          "color": "#00000030",
                        },
                      ]
                    }
                  >
                    Nickanme
                  </Text>
                  <TextInput
                    allowFontScaling={true}
                    autoCapitalize="none"
                    editable={true}
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    onFocus={[Function]}
                    placeholder="Please write your display name."
                    placeholderTextColor="#999999"
                    rejectResponderTermination={true}
                    secureTextEntry={false}
                    style={
                      Array [
                        Array [
                          Object {
                            "color": "#000000",
                            "flex": 1,
                            "fontSize": 14,
                            "fontWeight": "bold",
                            "paddingHorizontal": 8,
                            "paddingVertical": 12,
                          },
                          undefined,
                        ],
                        undefined,
                        false,
                      ]
                    }
                    underlineColorAndroid="transparent"
                    value=""
                  />
                </View>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "alignSelf": "stretch",
                      "flexDirection": "column",
                    },
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                      "marginHorizontal": 24,
                      "marginTop": 8,
                      "paddingVertical": 12,
                    },
                  ]
                }
                testID="container-id"
              >
                <View
                  style={
                    Array [
                      Array [
                        Object {
                          "alignItems": "center",
                          "alignSelf": "stretch",
                          "borderBottomColor": "#00000030",
                          "borderBottomWidth": 0.5,
                          "flexDirection": "row",
                          "justifyContent": "space-between",
                        },
                        Object {
                          "alignItems": "center",
                          "borderColor": "#00000030",
                          "borderWidth": 1,
                          "height": 200,
                          "marginTop": 8,
                          "paddingHorizontal": 20,
                          "width": "100%",
                        },
                      ],
                      false,
                      Object {
                        "borderBottomColor": "#00000030",
                        "borderColor": "#00000030",
                      },
                    ]
                  }
                >
                  <Text
                    style={
                      Array [
                        Array [
                          Object {
                            "color": "#999999",
                            "fontSize": 14,
                            "width": 100,
                          },
                          undefined,
                        ],
                        undefined,
                        Object {
                          "color": "#00000030",
                        },
                      ]
                    }
                  >
                    Introduction
                  </Text>
                  <TextInput
                    allowFontScaling={true}
                    autoCapitalize="none"
                    editable={true}
                    multiline={true}
                    numberOfLines={8}
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    onFocus={[Function]}
                    placeholder="Please introduce yourself."
                    placeholderTextColor="#999999"
                    rejectResponderTermination={true}
                    secureTextEntry={false}
                    style={
                      Array [
                        Array [
                          Object {
                            "color": "#000000",
                            "flex": 1,
                            "fontSize": 14,
                            "fontWeight": "bold",
                            "paddingHorizontal": 8,
                            "paddingVertical": 12,
                          },
                          Object {
                            "textAlignVertical": "center",
                          },
                        ],
                        Object {
                          "textAlignVertical": "center",
                        },
                        false,
                      ]
                    }
                    textAlignVertical="center"
                    underlineColorAndroid="transparent"
                    value=""
                  />
                </View>
              </View>
            </View>
          </View>
        </RCTScrollView>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row-reverse",
              "flexWrap": "wrap",
              "justifyContent": "center",
              "marginBottom": 60,
            }
          }
        >
          <View
            activeOpacity={0.7}
            delayPressIn={50}
            disabled={false}
            onPress={[MockFunction]}
          >
            <View
              onLayout={[Function]}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "alignSelf": "stretch",
                    "backgroundColor": "#393D7A",
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "paddingBottom": 8,
                    "paddingLeft": 12,
                    "paddingRight": 12,
                    "paddingTop": 8,
                  },
                  Object {
                    "backgroundColor": "#44D1A6",
                    "borderColor": "#FFFFFF",
                    "borderRadius": 6,
                    "borderWidth": 1,
                    "height": 44,
                    "marginBottom": 4,
                    "marginTop": 20,
                  },
                  false,
                  false,
                ]
              }
              testID="button-view"
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 14,
                    },
                    Object {
                      "alignSelf": "center",
                      "color": "#FFFFFF",
                      "fontSize": 14,
                      "paddingBottom": 2,
                      "textAlign": "center",
                      "width": 240,
                    },
                    false,
                  ]
                }
              >
                Update
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
                "marginHorizontal": 40,
                "marginTop": 24,
              }
            }
          >
            <View
              activeOpacity={0.7}
              delayPressIn={50}
              onPress={[MockFunction]}
            >
              <View
                onLayout={[Function]}
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "alignSelf": "stretch",
                      "backgroundColor": "#393D7A",
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "paddingBottom": 8,
                      "paddingLeft": 12,
                      "paddingRight": 12,
                      "paddingTop": 8,
                    },
                    Object {
                      "backgroundColor": "#FFFFFF",
                      "borderColor": "#000000",
                      "borderRadius": 6,
                      "borderWidth": 0.5,
                      "height": 44,
                    },
                    false,
                    undefined,
                  ]
                }
                testID="button-view"
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "white",
                        "fontSize": 14,
                      },
                      Object {
                        "alignSelf": "center",
                        "color": "#000000",
                        "fontSize": 14,
                        "textAlign": "center",
                        "width": 100,
                      },
                      undefined,
                    ]
                  }
                >
                  Go Back
                </Text>
              </View>
            </View>
            <View
              activeOpacity={0.7}
              delayPressIn={50}
              onPress={[MockFunction]}
            >
              <View
                onLayout={[Function]}
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "alignSelf": "stretch",
                      "backgroundColor": "#393D7A",
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "paddingBottom": 8,
                      "paddingLeft": 12,
                      "paddingRight": 12,
                      "paddingTop": 8,
                    },
                    Object {
                      "backgroundColor": "#FFFFFF",
                      "borderColor": "#FF7676",
                      "borderRadius": 6,
                      "borderWidth": 1,
                      "height": 44,
                      "marginLeft": 12,
                    },
                    false,
                    undefined,
                  ]
                }
                testID="button-view"
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "white",
                        "fontSize": 14,
                      },
                      Object {
                        "alignSelf": "center",
                        "color": "#FF7676",
                        "fontSize": 14,
                        "textAlign": "center",
                        "width": 100,
                      },
                      undefined,
                    ]
                  }
                >
                  Logout
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RNCSafeAreaView>
  </View>
</RNCSafeAreaProvider>
`;

exports[`Rendering renders without crashing 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <RNCSafeAreaView>
      <View
        onLayout={[Function]}
        style={
          Object {
            "paddingBottom": 0,
          }
        }
      >
        <RCTScrollView
          collapsable={false}
          contentContainerStyle={
            Object {
              "alignSelf": "stretch",
              "paddingHorizontal": 16,
              "paddingTop": 24,
            }
          }
          onGestureHandlerEvent={[Function]}
          onGestureHandlerStateChange={[Function]}
          style={
            Object {
              "alignSelf": "stretch",
            }
          }
        >
          <View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#FFFFFF",
                    "flexDirection": "column",
                    "justifyContent": "flex-start",
                  },
                ]
              }
            >
              <View>
                <View
                  onPress={[MockFunction]}
                  style={
                    Object {
                      "marginTop": 12,
                    }
                  }
                >
                  <Image
                    source={
                      Object {
                        "canInstrument": true,
                        "createTransformer": [Function],
                        "getCacheKey": [Function],
                        "process": [Function],
                      }
                    }
                    style={
                      Array [
                        Object {
                          "borderRadius": 60,
                          "height": 120,
                          "marginTop": 10,
                          "width": 120,
                        },
                      ]
                    }
                  />
                </View>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "alignSelf": "stretch",
                      "flexDirection": "column",
                    },
                    Object {
                      "marginHorizontal": 24,
                      "marginTop": 24,
                      "paddingHorizontal": 20,
                    },
                  ]
                }
                testID="container-id"
              >
                <View
                  style={
                    Array [
                      Array [
                        Object {
                          "alignItems": "center",
                          "alignSelf": "stretch",
                          "borderBottomColor": "#00000030",
                          "borderBottomWidth": 0.5,
                          "flexDirection": "row",
                          "justifyContent": "space-between",
                        },
                        Object {
                          "height": 54,
                        },
                      ],
                      false,
                      Object {
                        "borderBottomColor": "#00000030",
                        "borderColor": "#00000030",
                      },
                    ]
                  }
                >
                  <Text
                    style={
                      Array [
                        Array [
                          Object {
                            "color": "#999999",
                            "fontSize": 14,
                            "width": 100,
                          },
                          undefined,
                        ],
                        undefined,
                        Object {
                          "color": "#00000030",
                        },
                      ]
                    }
                  >
                    Nickanme
                  </Text>
                  <TextInput
                    allowFontScaling={true}
                    autoCapitalize="none"
                    editable={true}
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    onFocus={[Function]}
                    placeholder="Please write your display name."
                    placeholderTextColor="#999999"
                    rejectResponderTermination={true}
                    secureTextEntry={false}
                    style={
                      Array [
                        Array [
                          Object {
                            "color": "#000000",
                            "flex": 1,
                            "fontSize": 14,
                            "fontWeight": "bold",
                            "paddingHorizontal": 8,
                            "paddingVertical": 12,
                          },
                          undefined,
                        ],
                        undefined,
                        false,
                      ]
                    }
                    underlineColorAndroid="transparent"
                    value=""
                  />
                </View>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "alignSelf": "stretch",
                      "flexDirection": "column",
                    },
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                      "marginHorizontal": 24,
                      "marginTop": 8,
                      "paddingVertical": 12,
                    },
                  ]
                }
                testID="container-id"
              >
                <View
                  style={
                    Array [
                      Array [
                        Object {
                          "alignItems": "center",
                          "alignSelf": "stretch",
                          "borderBottomColor": "#00000030",
                          "borderBottomWidth": 0.5,
                          "flexDirection": "row",
                          "justifyContent": "space-between",
                        },
                        Object {
                          "alignItems": "center",
                          "borderColor": "#00000030",
                          "borderWidth": 1,
                          "height": 200,
                          "marginTop": 8,
                          "paddingHorizontal": 20,
                          "width": "100%",
                        },
                      ],
                      false,
                      Object {
                        "borderBottomColor": "#00000030",
                        "borderColor": "#00000030",
                      },
                    ]
                  }
                >
                  <Text
                    style={
                      Array [
                        Array [
                          Object {
                            "color": "#999999",
                            "fontSize": 14,
                            "width": 100,
                          },
                          undefined,
                        ],
                        undefined,
                        Object {
                          "color": "#00000030",
                        },
                      ]
                    }
                  >
                    Introduction
                  </Text>
                  <TextInput
                    allowFontScaling={true}
                    autoCapitalize="none"
                    editable={true}
                    multiline={true}
                    numberOfLines={8}
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    onFocus={[Function]}
                    placeholder="Please introduce yourself."
                    placeholderTextColor="#999999"
                    rejectResponderTermination={true}
                    secureTextEntry={false}
                    style={
                      Array [
                        Array [
                          Object {
                            "color": "#000000",
                            "flex": 1,
                            "fontSize": 14,
                            "fontWeight": "bold",
                            "paddingHorizontal": 8,
                            "paddingVertical": 12,
                          },
                          Object {
                            "textAlignVertical": "center",
                          },
                        ],
                        Object {
                          "textAlignVertical": "center",
                        },
                        false,
                      ]
                    }
                    textAlignVertical="center"
                    underlineColorAndroid="transparent"
                    value=""
                  />
                </View>
              </View>
            </View>
          </View>
        </RCTScrollView>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row-reverse",
              "flexWrap": "wrap",
              "justifyContent": "center",
              "marginBottom": 60,
            }
          }
        >
          <View
            activeOpacity={0.7}
            delayPressIn={50}
            disabled={false}
            onPress={[MockFunction]}
          >
            <View
              onLayout={[Function]}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "alignSelf": "stretch",
                    "backgroundColor": "#393D7A",
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "paddingBottom": 8,
                    "paddingLeft": 12,
                    "paddingRight": 12,
                    "paddingTop": 8,
                  },
                  Object {
                    "backgroundColor": "#44D1A6",
                    "borderColor": "#FFFFFF",
                    "borderRadius": 6,
                    "borderWidth": 1,
                    "height": 44,
                    "marginBottom": 4,
                    "marginTop": 20,
                  },
                  false,
                  false,
                ]
              }
              testID="button-view"
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 14,
                    },
                    Object {
                      "alignSelf": "center",
                      "color": "#FFFFFF",
                      "fontSize": 14,
                      "paddingBottom": 2,
                      "textAlign": "center",
                      "width": 240,
                    },
                    false,
                  ]
                }
              >
                Update
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
                "marginHorizontal": 40,
                "marginTop": 24,
              }
            }
          >
            <View
              activeOpacity={0.7}
              delayPressIn={50}
              onPress={[MockFunction]}
            >
              <View
                onLayout={[Function]}
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "alignSelf": "stretch",
                      "backgroundColor": "#393D7A",
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "paddingBottom": 8,
                      "paddingLeft": 12,
                      "paddingRight": 12,
                      "paddingTop": 8,
                    },
                    Object {
                      "backgroundColor": "#FFFFFF",
                      "borderColor": "#000000",
                      "borderRadius": 6,
                      "borderWidth": 0.5,
                      "height": 44,
                    },
                    false,
                    undefined,
                  ]
                }
                testID="button-view"
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "white",
                        "fontSize": 14,
                      },
                      Object {
                        "alignSelf": "center",
                        "color": "#000000",
                        "fontSize": 14,
                        "textAlign": "center",
                        "width": 100,
                      },
                      undefined,
                    ]
                  }
                >
                  Go Back
                </Text>
              </View>
            </View>
            <View
              activeOpacity={0.7}
              delayPressIn={50}
              onPress={[MockFunction]}
            >
              <View
                onLayout={[Function]}
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "alignSelf": "stretch",
                      "backgroundColor": "#393D7A",
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "paddingBottom": 8,
                      "paddingLeft": 12,
                      "paddingRight": 12,
                      "paddingTop": 8,
                    },
                    Object {
                      "backgroundColor": "#FFFFFF",
                      "borderColor": "#FF7676",
                      "borderRadius": 6,
                      "borderWidth": 1,
                      "height": 44,
                      "marginLeft": 12,
                    },
                    false,
                    undefined,
                  ]
                }
                testID="button-view"
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "white",
                        "fontSize": 14,
                      },
                      Object {
                        "alignSelf": "center",
                        "color": "#FF7676",
                        "fontSize": 14,
                        "textAlign": "center",
                        "width": 100,
                      },
                      undefined,
                    ]
                  }
                >
                  Logout
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RNCSafeAreaView>
  </View>
</RNCSafeAreaProvider>
`;
